src/python/dxpy/scripts/dx.py

```diff
@@ -3832,9 +3832,14 @@ def get_archival_paths(args):
             possible_objects.add(p[-1])
 
         # get project ID
+        if not possible_projects:
+            if not dxpy.PROJECT_CONTEXT_ID:
```

Comment:
What happens with 
```
 dx select p2; 
 dx archive p1:f1 f2
``` 
Do we error out because f2 should be resolved as p2:f2 and p2 != p1?

Reply:
Yes. In this scenario, since we have project id given in some paths, here the `possible_projects = {'f1',''}` and `''` is later interpreted as p2.

Reply:
my tests show that the above test case needs fixing.